state/unit.go

```diff
@@ -130,7 +130,9 @@ func (u *Unit) ConfigSettings() (charm.Settings, error) {
 	}
 	result := chrm.Config().DefaultSettings()
 	for name, value := range settings.Map() {
-		result[name] = value
+		if value != nil {
```

Comment:
This *shouldn't* happen right? IIANM it'll only happen because there's no upgrade step to remove them. I'd really prefer we had an upgrade step instead, and kept the model clean.